src/material-experimental/mdc-slider/slider.spec.ts

```diff
@@ -650,6 +797,22 @@ function slideToValue(slider: MatSlider, value: number, thumbPosition: Thumb, is
   dispatchPointerOrTouchEvent(sliderElement, PointerEventType.POINTER_UP, endX, endY, isIOS);
 }
 
+/**
+ * Mimics changing the slider value using arrow keys.
+ *
+ * Dispatching keydown events on inputs do not trigger value changes. Thus, to mimic this behavior,
+ * we manually change the slider inputs value and then dispatch a change event (which is what the
+ * MDC Foundation is listening for & how it handles these updates).
+ */
+function changeValueUsingArrowKeys(slider: MatSlider, arrow: ArrowKey, thumbPosition: Thumb) {
+  const input = slider._getInput(thumbPosition);
+  const value = arrow === ArrowKey.RIGHT
+    ? input.value + slider.step
+    : input.value - slider.step;
+  input._hostElement.value = value.toString();
+  input._hostElement.dispatchEvent(new Event('change'));
```

Comment:
Rather than calling `dispatchEvent` yourself, you should use `dispatchEvent` or `dispatchFakeEvent` from `@angular/cdk/testing/private`.